Marc branch

ScottCampit opened the pull request:
Code review 1: Checking @marcdimeo 's code formatting for production.

marcdimeo committed:
<New python code to find all CCLE gene data>

marcdimeo committed:
<Trying to add my new pyscript to github>

marcdimeo committed:
<I am still trying to add pyscript>

marcdimeo committed:
Merge branch 'master' of https://github.com/ScottCampit/egem into MarcBranch
This is an update of MarcBranch with everything from master

Commit:
Merge branch 'master' into MarcBranch

marcdimeo committed:
<This code serves as a new script in order to group celllines into tissues which will be used for histone correlation studies>

marcdimeo committed:
<Updating my newest python code>

marcdimeo committed:
<Trying to update the python code so that it includes tissue analysis for histone correlation>

Commit:
Merge branch 'master' into MarcBranch

marcdimeo committed:
Trying to update my filesMerge branch 'master' of https://github.com/ScottCampit/egem into MarcBranch

marcdimeo committed:
<Updating new python code too include a tissue analysis which will be used for histone correlation in the future>

marcdimeo committed:
<Updating MarcBranch to include tissue analysis code>

marcdimeo committed:
Merge branch 'MarcBranch' of https://github.com/ScottCampit/egem into MarcBranch

marcdimeo committed:
<Updating all of the python files to include heatmaps and all genes from CCLE. This also includes adding csv files and figures and changing to the lab github>

marcdimeo committed:
<Add in new RECON1 gene set with heatmap which needs updpating>

marcdimeo committed:
<Updating MarcBranch to include clustergram code which is used for all metabolic genes of Recon1>

marcdimeo committed:
<Pushing new code to add clustergram for Recon1 genes, this can be used to visualize all metabolic genes>

marcdimeo committed:
<Additional code to add clustergram data so that all Recon1 genes to better visualize the data>

Commit:
added new functional python scripts

marcdimeo committed:
<Adding new pyscript called master which is a cleaner version of the CCLE code. It is not complete but contains all necessary functions>

ScottCampit commented on matlab/scripts/correlation_value.m:
```diff
@@ -9,6 +9,8 @@
 %% Load in files
```
Not sure what this code is doing - I don't see it calculating a correlation value, which is what the file name suggests. You should rename the file name, and clean up the rest of the script.

marcdimeo replied:
Thats the old one (one of the ones I might just delete)

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
 
```
You need to finish the code with `return common` to actually output the variable common.

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -20,16 +20,20 @@
-#Used to find common celllines
-def common_celllines(a,b):
+
+def common(a,b):
```
Name your input variables with better descriptions (ie list_a, list_b)

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -20,16 +20,20 @@
-def common_celllines(a,b):
+
+def common(a,b):
+    """ This functions purpose is to find common elements of lists by converting
```
Example of how you should write all your docstrings:

"""
common finds all common elements between two lists

INPUTS:
  * a: a list
  * b: another list

OUTPUTS:
  * common: a set containing common elements between list a and list b
"""

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
+    
+    The format for inputting matrix should have columns correspond to the celllines.
+    """
+    Rmat = (len(gene_list), len(histone_list))
```
Unecessary - call Matrix to get Rmat and Pmat

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
+            j = j+1
+        i = i+1
+        
+    Rdf = pd.DataFrame(Rmat)
```
It looks like you can merge Matrix and pearson_dfs into a single function. This seems redundant.

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
+
+
+    
+def Remove(duplicate): 
```
Have a more descriptive function name ie `remove_duplicates`

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
 
 #Used to create correlation matrix
 def Matrix(gene_list, histone_list, data1, data2):
```
name your functions better. matrix does not tell me what this function is doing.

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
+            final_list.append(num) 
+    return final_list
+
+def Clustermap(data, size, colour, method, metric):
```
Move to `vis.py` because this is a visualization function

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
+    ax = sns.clustermap(data, cmap = colour, method = method, metric= metric)
+    plt.show
+    
+def PlotlyHeat(df, size, title, xaxis, yaxis):
```
same - move to vis

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
+    matrix = matrix.set_index('Genes')
+    return matrix
 
 """
```
I don't know what this code is for. If you're cleaning up the dataset, you may want to move this to another script.

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -219,7 +325,7 @@ def Heatmap(data, size, colour, title):
 REORGANIZING DATA
 """
```
Okay, this looks like you're running all the functions. This can be in a separate script called 'exploratory_correlation_analysis.py` or something to that effect.

ScottCampit commented on python/CCLE_data.py:
```diff
@@ -219,7 +325,7 @@ def Heatmap(data, size, colour, title):
 """
 #Finding common cell lines
-h3_ccle_cellline = common_celllines(cellline_list, h3_celllines) 
+h3_ccle_cellline = common(cellline_list, h3_celllines) 
```
I'm seeing similar structures. turn into function

ScottCampit commented on python/master.py:
```diff
@@ -0,0 +1,321 @@
+FUNCTIONS
+"""
+
+def read_csv_df(file):
```
Example of docstring. Apply to all functions:

"""
read_csv_df takes in the path to a file with whitespace delimiters and replaces zero values with NaN.

INPUT:
  * file: path to the file

OUTPUT:
  * data: data with empty values as NaN.
"""

ScottCampit commented on python/master.py:
```diff
@@ -0,0 +1,321 @@
+        i = i+1
+    return Rmat, Pmat
+    
+def pearson_dfs(gene_list, histone_list, data1, data2):
```
This code seems redundant with the `matrix` function, which is also not a great way to name the function, since matrix doesn't tell me anything about the functionality. You may want to merge the two functions together. 

If you really need to keep the numpy array, make another argument called dtype, and make it a conditional, such that if dtype='numpy', return numpy array, else return pandas dataframe.

ScottCampit commented on python/master.py:
```diff
@@ -0,0 +1,321 @@
+#import plotly.graph_objects as go
+
+"""
+FUNCTIONS
```
Make sure your code follows this style guide: https://realpython.com/python-pep8/

ScottCampit commented on python/master.py:
```diff
@@ -0,0 +1,321 @@
+    return matrix
+
+"""
+CODE TO GET CCLE NORMALIZED
```
If this code utilizes your functions, you should put this in a separate script (ie run_epigenetic_correlation.py) and call the functions in this module by using import master.

marcdimeo replied:
How much would you like me to run in this file because I can run all the way through creating the heat maps to just simply creating 3 data frames (LeRoy, H3, and CCLE) or I can go anywhere in between I just dont know what is necessary

ScottCampit commented on python/master.py:
```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env python3
```
Great start. This file seems to have a lot of utility functions for your histone analysis, so I would suggest renaming the file "histone_utility.py" or something along those lines. A good file name should describe its contents well.

ScottCampit commented on python/master.py:
```diff
@@ -0,0 +1,321 @@
+
+
+def heatmap(data, ylabels, size, colour, title):
+    """Using dataframes based on the pearson_dfs fucntion, this function will create a simple seaborn heatmap. By using a dataframe the xlabels will be automatically created from the dataframes columns
```
Put these in the vis.py script and add your name as an author.

marcdimeo replied:
done

ScottCampit commented on python/master.py:
```diff
@@ -0,0 +1,321 @@
+
+
+    
+def plotlyheat(df, size, title, xaxis, yaxis):
```
Put these in the vis.py script and add your name as an author.

marcdimeo replied:
done

ScottCampit commented on python/pearson_matrix.py:
```diff
@@ -0,0 +1,96 @@
+"""
+Write a function to load your data. I'll be using a random dataset with approximately the same size as the CCLE H3 relative values proteomics dataset. The histone marks and genes are the rows, while the columns are the cell line values. You can perform numpy calculations on pandas dataframes, but you will have to figure out how to do that on your own.
```
If you don't need this script anymore, you can delete it. If you want to keep it, you can edit your .gitignore file in the top of the egem folder and type in the path to the file. That will cause git to not add this file when pushing.

marcdimeo replied:
Going to delete soon just at the end to be sure

ScottCampit commented on python/tissue_analysis.py:
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
```
If you don't need this script anymore, you can delete it. If you want to keep it, you can edit your .gitignore file in the top of the egem folder and type in the path to the file. That will cause git to not add this file when pushing.

marcdimeo replied:
Going to delete soon just at the end to be sure

ScottCampit reviewed (commented):
Good first start. Comments are in specific files.

marcdimeo committed:
<Convert master into two seperate documents histone_utility and run_epigenetic_correlation. Add better descriptions for functions>

Commit:
SC_edits for histone_utility.py

@ScottCampit made the following edits:
  * read_csv_df: edited the input variable from `file` to `csv_file`.
  * changed name of `columns_remove` to `remove_columns`
    * changed `df` variable name to `data` to keep consistent with previous function
  * changed name of `remove` to `remove_duplicate_entries` to make function name clearer.
    * changed `list_of_items` name to `items_to_remove`, to make it more clear that we're removing these items.

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    """This serves as a way to convert gene ids into gene  
+    symbols which is much easier to work with. 
+    
+    It is important to note that the gene_list may require manipulation 
```
What does this mean? This description is too vague. Clarify what this manipulation is, how mygene will work differently, and what the scope is.

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    It is important to note that the gene_list may require manipulation 
+    as mygene works differently for each scope you use. 
+    
+    The gene_list may require some editing.
```
how?

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    The gene_list may require some editing.
+    
+    The output is a list of dictionary which contain ids and gene symbols. 
+    Further manipulation is required regarding removing any duplicate IDs.
```
what manipulation?

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    
+    INPUT
+    
+        gene_list: list of genes which you would like to convert into desired 
```
what's the format? gene symbol, ensemble id, uniprot id?

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+        species: type of species you are looking at
+    
+    OUTPUT
+        ginfo: list of dictionaries containing gene ids, 
```
what is the scores? what's the query?

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+
+
+def tissue_dict(df):
+    """Using a df with number indexing and not gene symbol 
```
run on sentence. please write to make it much more clearer.

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    indexing (so not pearson_dfs) this will create a dictionary that will 
+    match celllines to their tissue type.
+    
+    This has tissue types split with an underscore so it will remove 
```
I read this as it will only remove the underscore. be more descriptive.

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+
+
+def normalize(df, delete_column, cell_list):
+    """Takes in a dataframe with columns being cell lines 
```
what is the type of normalization? what is the expected output?

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    (with the exception of a column being genes or gene_ids) and normalizes 
+    the expression values.
+    
+    This will delete the desired column and then normalize the values and 
```
need to be more clear. i'm guessing this function returns a dataframe of the same size back out, not the dataframe as it originally looked being inputted.

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    
+    INPUT
+    
+        df: dataframe with columns of cell lines (with the exception of a 
```
?

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+
+    temp_df = df.copy()
+    
+    delete = temp_df[delete_column]
```
can be simplified into 1 line: temp.pop([delete_column])

marcdimeo replied:
im not sure what you mean by this but I changed the code becuase delete and delete_column are different in this case.

ScottCampit replied:
`pop` removes the object from the dataframe: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.pop.html

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    del temp_df[delete_column]
+    
+    x = temp_df.values.astype(float)
+    min_max_scaler = preprocessing.MinMaxScaler()
```
mention that you're performing a min-max transformation in the docstring.

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    return df_normalized
+
+
+def common(a,b):
```
change variable names to list_a and list_b

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    return common 
+    
+def pearson_dfs(gene_list, histone_list, data1, data2, dtype):
+    """This functions purpose is to create two dataframes. 
```
the purpose is not to create two dataframes: it is to compute the column-wise pearson correlation coefficient and p-value for each element in an array.

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    INPUT
+        gene_list: list of genes which correlation study will be performed
+        histone_list: list of histones which correlation study will be performed
+        data1: dataframe containing genes which you would like to compare
```
name gene_data

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+        gene_list: list of genes which correlation study will be performed
+        histone_list: list of histones which correlation study will be performed
+        data1: dataframe containing genes which you would like to compare
+        data2: dataframe containing histones which you would like to compare
```
name histone_data

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+        histone_list: list of histones which correlation study will be performed
+        data1: dataframe containing genes which you would like to compare
+        data2: dataframe containing histones which you would like to compare
+        dtype: numpy or df, numpy returns a matrix, and df returns a dataframe
```
add quotes. make default argument numpy array.

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+    function and the name will correspond to the type of tissue wishing 
+    to be analyzed. The output is a dataframe which can be graphed.
+    
+    This may not give you the best idea of tissue analysis as most datasets 
```
?

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,342 @@
+        return Rdf, Pdf
+    
+    
+def tissue_analysis(dictionary, common_cellline, histone_list, gene_list, 
```
rename `dictionary` to `cell_line_dict`

ScottCampit reviewed (commented):
Another iteration needed: see comments and address points in this script.

marcdimeo committed:
<Updating and continuation of cleaning up the code>

Commit:
Example of vectorized function in `remove_columns`

Original code:
```
 for columns in list_columns:
        del data[columns]
```

new code:
```
    data = data.drop(columns, axis=1)
```

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,355 @@
+        
+    """
+    final_list = [] 
+    for num in items_to_remove: 
```
Need to vectorize: https://pandas.pydata.org/pandas-docs/version/0.17/generated/pandas.DataFrame.drop_duplicates.html

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,355 @@
+    It is important to note that the gene_list may require manipulation 
+    as mygene works differently for each scope you use. The manipulation will 
+    depend on the scope being used. For example 'ensembl.gene' ids look similar
+    to ENSG00000000003 however they can have decimals (ENSG00000000003.3)
```
This sounds like a simple fix. Can you write a conditional where if they choose ensembl gene identifiers, it turns that float value to an int?

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,355 @@
+    for gene in gene_list:
+        j = 0
+        for histone in histone_list:
+            gene_data.loc[gene] = gene_data.loc[gene].fillna(method='ffill')
```
This part can actually be taken out of the for loop and can be vectorized.

ScottCampit replied:
gene_data = gene_data.fillna(method='ffill') should work

marcdimeo replied:
I think I fixed this one

ScottCampit commented on python/histone_utility.py:
```diff
@@ -0,0 +1,355 @@
+    
+    df1 = data1.copy()
+    df2 = data2.copy()
+    for celllines in common_cellline:
```
Can be vectorized using iloc

marcdimeo replied:
I think I need another hint on this one cause I cant figure out the logic like I understand what youre saying but like im confused on how I would do it

ScottCampit replied:
look up "deleted columns based on list pandas" on google. there are many ways to tackle this problem.

marcdimeo committed:
<Beginning to vectorize the code in histone utility to make it faster>

marcdimeo committed:
<Fixed tissue_analysis so that it is faster and updated vis.py to include chart_studio>

Commit:
Merge branch 'master' into MarcBranch
